topdown/query.go

```diff
@@ -66,11 +67,12 @@ type Builtin struct {
 // NewQuery returns a new Query object that can be run.
 func NewQuery(query ast.Body) *Query {
 	return &Query{
-		query:        query,
-		genvarprefix: ast.WildcardPrefix,
-		indexing:     true,
-		earlyExit:    true,
-		external:     newResolverTrie(),
+		query:          query,
+		genvarprefix:   ast.WildcardPrefix,
+		indexing:       true,
+		earlyExit:      true,
+		external:       newResolverTrie(),
+		ndBuiltinCache: builtins.NDBCache{},
```

Comment:
🔍 Does this get used anywhere? Tested? I'm wondering if we should default to something non-nil here... after all, we're trying to have users opt into this, and isn't it auto-enabled when we set it to something here?

Reply:
I suspect I set it to something non-nil here because leaving it nil resulted in a panic during testing. I'll try setting it to a nil value, and will report back what I find.

Reply:
Doesn't seem to cause issues when removed, now that the cache is properly opt-in. :+1: